weaviate/collection/collection_model.py

```diff
@@ -138,12 +138,24 @@ def reference_add_many(self, from_property: str, refs: List[BatchReference]) ->
 
 
 class _GRPCWrapper(Generic[Model]):
-    def __init__(self, connection: Connection, name: str, model: Type[Model]):
+    def __init__(
+        self,
+        collection: "CollectionObjectModel[Model]",
+        connection: Connection,
+        name: str,
+        model: Type[Model],
```

Comment:
Since `connection`, `name`, and `model` all exist on `collection` (and are indeed sourced from it), would it be better to have only `collection` here that provides those vars?

Reply:
I added properties to collection for name and model. Didn't want to add one for connection as this is something that users should not see. I don't like using protected variables in other classes, so I'm still passing connection